packages/replica-healthcheck/src/healthcheck-server.ts

```diff
@@ -91,17 +112,63 @@ export class HealthcheckServer {
     return server
   }
 
+  initTxLatencyCheck = () => {
+    // Check latency for every Monday
+    cron.schedule('0 0 * * 1', this.runTxLatencyCheck)
+  }
+
+  runTxLatencyCheck = async () => {
+    const wallet1 = new Wallet(
+      this.options.txWriteOptions.wallet1PrivateKey,
+      this.replicaProvider
+    )
+    const wallet2 = new Wallet(
+      this.options.txWriteOptions.wallet2PrivateKey,
+      this.replicaProvider
+    )
+
+    // Send funds between the 2 addresses
+    const res1 = await this.getLatencyForSend(wallet1, wallet2)
+    this.logger.info('Sent transaction from wallet1 to wallet2', {
+      latencyMs: res1.latencyMs,
+      status: res1.status,
+    })
+    const res2 = await this.getLatencyForSend(wallet2, wallet2)
+    this.logger.info('Sent transaction from wallet2 to wallet1', {
+      latencyMs: res2.latencyMs,
+      status: res2.status,
+    })
+  }
+
+  getLatencyForSend = async (
+    from: Wallet,
+    to: Wallet
+  ): Promise<{
+    latencyMs: number
+    status: number
+  }> => {
+    const fromBal = await from.getBalance()
```

Comment:
Perhaps it would be a good idea to have a check that the balance is not 0 here

Reply:
great catch!